contracts/interchain-token-service/tests/utils/mod.rs

```diff
@@ -1,14 +1,56 @@
+use std::fmt::Debug;
+
 use axelar_gas_service::{AxelarGasService, AxelarGasServiceClient};
 use axelar_gateway::testutils::{setup_gateway, TestSignerSet};
 use axelar_gateway::AxelarGatewayClient;
+use axelar_soroban_std::events::Event;
+use axelar_soroban_std::impl_event_testutils;
 use axelar_soroban_std::types::Token;
 use interchain_token_service::{InterchainTokenService, InterchainTokenServiceClient};
-use soroban_sdk::{testutils::Address as _, token::StellarAssetClient, Address, Env, String};
+use soroban_sdk::Topics;
+use soroban_sdk::{
+    testutils::Address as _, token::StellarAssetClient, Address, Bytes, BytesN, Env, IntoVal,
+    String, Symbol, Val,
+};
 
 pub const HUB_CHAIN: &str = "hub_chain";
 
 const INTERCHAIN_TOKEN_WASM_HASH: &[u8] = include_bytes!("../testdata/interchain_token.wasm");
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct InterchainTransferReceivedEvent {
```

Comment:
The event should be defined within the contract crate and emitted within the contract. It's not meant to just be defined in the test. See existing usage